hydra-tx/testlib/Test/Hydra/Tx/Gen.hs

```diff
@@ -112,17 +173,51 @@ genOneUTxOFor vk = do
   output <- scale (const 1) $ genOutput vk
   pure $ UTxO $ Map.singleton input output
 
--- | Generate 'Babbage' era 'UTxO', which may contain arbitrary assets in
--- 'TxOut's addressed to public keys *and* scripts. NOTE: This is not reducing
--- size when generating assets in 'TxOut's, so will end up regularly with 300+
--- assets with generator size 30. NOTE: The Arbitrary TxIn instance from the
--- ledger is producing colliding values, so we replace them.
-genUTxOAlonzo :: Gen UTxO
-genUTxOAlonzo = do
-  utxoMap <- Map.toList . Ledger.unUTxO <$> arbitrary
-  fmap UTxO.fromPairs . forM utxoMap $ \(_, o) -> do
-    i <- arbitrary
-    pure (i, fromLedgerTxOut o)
+-- | Generate "simplified" UTXO, ie. without some of the complexities required
+-- for backward-compatibility and obscure features.
+genUTxOWithSimplifiedAddresses :: Gen UTxO
+genUTxOWithSimplifiedAddresses =
+  UTxO.fromPairs <$> listOf genEntry
+ where
+  genEntry = (,) <$> genTxIn <*> genTxOut
+
+-- * Others
+
+instance Arbitrary AssetName where
+  arbitrary = AssetName . BS.take 32 <$> arbitrary
+
+genKeyPair :: Gen (VerificationKey PaymentKey, SigningKey PaymentKey)
+genKeyPair = do
+  sk <- genSigningKey
+  pure (getVerificationKey sk, sk)
+
+genValue :: Gen Value
+genValue = fmap ((lovelaceToValue $ Coin 10_000_000) <>) (scale (`div` 10) $ fromLedgerValue <$> arbitrary)
+
+genVerificationKey :: Gen (VerificationKey PaymentKey)
+genVerificationKey = getVerificationKey <$> genSigningKey
+
+-- | NOTE: See note on 'mkVkAddress' about 'NetworkId'.
+genAddressInEra :: NetworkId -> Gen AddressInEra
+genAddressInEra networkId =
+  mkVkAddress networkId <$> genVerificationKey
+
+-- TODO: Enable arbitrary datum in generators
```

Comment:
Seems like this TODO should be already dealt with here and we should enable datum generation in this PR already since we are adding datum information to tx outputs?